run/authentication/auth.js

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const requestServiceToken = async (receivingServiceURL = 'https://SERVICE_NAME-HASH-run.app') => {
```

Comment:
Not sure I follow why this function exists? If we're running in an environment that can mint a token, let's write an integration test that mints the token. We can split the request function to something that returns the token from metadata so it can be tested without making an HTTP request, or we can keep this all in one but use request mocking to inspect and validate the token: https://github.com/sindresorhus/got#testing